### PR TITLE
Modifications in  Data Team Helper

### PR DESCRIPTION
**Pull request by sruthymathew123**
Implementing the suggested features for Data Team Helper in LORIS module review meeting.

- Adding Site & Project filter to data team helper
- Adding column PSCID to react table.
 
Also, fixing some existing bugs;
Filter options not holding/displaying the values(for instrument dropdown) in their fields once they are applied.

### CODE REVIEW

File: modules/data_team_helper/php/NDB_Form_data_team_helper.class.inc [hunk 1]
```diff
@@ -180,22 +217,30 @@ class NDB_Form_Data_Team_Helper extends NDB_Form
...
+        $project=null
+    ) {
 
         $DB      =& Database::singleton();
         $qparams = array();
 
         $select = "SELECT fb.FeedbackID, fb.CandID, fb.SessionID,
                       fb.CommentID, fb.Feedback_level, f.Test_name, fb.FieldName,
-                      tn.Full_name, s.Visit_label FROM feedback_bvl_thread AS fb
+                      tn.Full_name, s.Visit_label, s.PSCID
```
Comment (johnsaigle) [inline modules/data_team_helper/php/NDB_Form_data_team_helper.class.inc]: The module is broken for me on this line. `s.PSCID` doesn't seem to work.
screen shot 2017-05-04 at 11 52 12 am
screen shot 2017-05-04 at 11 52 54 am

File: modules/data_team_helper/php/NDB_Form_data_team_helper.class.inc [hunk 2]
```diff
@@ -127,12 +132,35 @@ class NDB_Form_Data_Team_Helper extends NDB_Form
...
+                $this->tpl_data['projectList'] = $projectList;
+            }
 
             $incompleteCandidates = $this->
                  getIncompleteCandidates(
                      $instrument,
                      $_REQUEST['visit_label'],
-                     $candID
+                     $candID,
+                     $_REQUEST['site'],
```
Comment (johnsaigle) [inline modules/data_team_helper/php/NDB_Form_data_team_helper.class.inc]: Before using any values from `$_REQUEST` they should be sanitized for malicious input, especially Javascript.

Since these values are being submitted directly into the page they are vulnerable to a [Reflected XSS attack](https://www.hacksplaining.com/exercises/xss-reflected). 

Going forward, we should prevent against this by [rejecting malformed input](https://github.com/aces/Loris/pull/2775), or, less preferably, sanitizing the output using e.g. [htmlspecialchars()](http://php.net/manual/en/function.htmlspecialchars.php).

**Comment by sruthymathew123** [comment]
@johnsaigle This module was overridden for ibis and above feature is working well on IBIS. In session table(ibis), it has the column PSCID. I don't know the case why it is not there in LORIS schema.

**Comment by johnsaigle** [comment]
Do you have a non-IBIS environment to test this on? Just to verify that it will work for other LORISes. If not, you'll need to issue a patch for this to work, I think.

**Comment by sruthymathew123** [comment]
@johnsaigle Discussed the thing with Greg and we both don't know why a PCSID column is there in session table of ibis schema which is not there in LORIS schema. 

Depending for PSCID on session table is not reliable as we are always using candidate table to get PSCID. I will alter the query with a join. Thanks.

**Comment by sruthymathew123** [comment]
@johnsaigle @gluneau  Query updated.

**Comment by sruthymathew123** [comment]
@johnsaigle Here nothing seems to be passed via URL, I think.

**Comment by sruthymathew123** [comment]
@johnsaigle Can you have a look again?

**Comment by johnsaigle** [comment]
I wouldn't say the problem is solved but I won't block this PR given that the problem I listed is systemic in the codebase; instead, we will try to refactor things more securely in a later release.

**Comment by sruthymathew123** [comment]
@johnsaigle Security concerns regarding this PR is fixed as you suggested. I sanitized all the input coming from the superglobal request. Can you have some time to review the code again?

**Comment by sruthymathew123** [comment]
ok.

**Comment by sruthymathew123** [comment]
@johnsaigle Done.

**Comment by sruthymathew123** [comment]
@ZainVirani Can you test this?

File: modules/data_team_helper/php/NDB_Form_data_team_helper.class.inc [hunk 3]
```diff
@@ -286,12 +346,16 @@ class NDB_Form_Data_Team_Helper extends NDB_Form
      * @param unknown_type $test_name   the value of test name
      * @param string       $visit_label the value of visit label
      * @param string       $candidate   the value of candidate
+     * @param null         $site        the value of site
+     * @param null         $project     the value of project
```
Comment (driusan) [inline modules/data_team_helper/php/NDB_Form_data_team_helper.class.inc]: if the type for the variables `$site` and `$project` are null, why are they included at all?

File: modules/data_team_helper/php/NDB_Form_data_team_helper.class.inc [hunk 4]
```diff
@@ -413,12 +487,16 @@ class NDB_Form_Data_Team_Helper extends NDB_Form
...
```
Comment (driusan) [inline modules/data_team_helper/php/NDB_Form_data_team_helper.class.inc]: same

File: modules/data_team_helper/php/NDB_Form_data_team_helper.class.inc [hunk 5]
```diff
@@ -466,11 +552,17 @@ class NDB_Form_Data_Team_Helper extends NDB_Form
...
```
Comment (driusan) [inline modules/data_team_helper/php/NDB_Form_data_team_helper.class.inc]: same

File: modules/data_team_helper/php/NDB_Form_data_team_helper.class.inc [hunk 6]
```diff
@@ -491,7 +583,7 @@ class NDB_Form_Data_Team_Helper extends NDB_Form
 
         $where =" WHERE s.Active='Y'
             AND s.Current_stage <> 'Recycling Bin'
-            AND f.CommentID NOT LIKE '%DDE%'
+            AND f.CommentID NOT LIKE '%DDE%'
```
Comment (driusan) [inline modules/data_team_helper/php/NDB_Form_data_team_helper.class.inc]: GitHub says that there's some unnecessary whitespace added to the end of this line.

File: modules/data_team_helper/php/NDB_Form_data_team_helper.class.inc [hunk 7]
```diff
@@ -516,11 +616,17 @@ class NDB_Form_Data_Team_Helper extends NDB_Form
      * @param unknown_type $test_name   the value of test name
      * @param string       $visit_label the value of visit label
      * @param string       $candidate   the value of candidate
+     * @param null         $site        the value of site
+     * @param null         $project     the value of project
```
Comment (driusan) [inline modules/data_team_helper/php/NDB_Form_data_team_helper.class.inc]: same "null" issue..

**Comment by sruthymathew123** [comment]
rebased

**Comment by sruthymathew123** [comment]
@driusan Can you have a look again?

File: modules/data_team_helper/php/NDB_Form_data_team_helper.class.inc [hunk 8]
```diff
@@ -180,22 +219,31 @@ class NDB_Form_Data_Team_Helper extends NDB_Form
      * @param null $test_name   the value of test name
      * @param null $candID      the value of candidate ID
      * @param null $visit_label the value of visit label
+     * @param null $site        the value of site
+     * @param null $project     the value of project
```
Comment (driusan) [inline modules/data_team_helper/php/NDB_Form_data_team_helper.class.inc]: missed two

File: modules/data_team_helper/php/NDB_Form_data_team_helper.class.inc [hunk 9]
```diff
@@ -286,12 +346,16 @@ class NDB_Form_Data_Team_Helper extends NDB_Form
      * @param unknown_type $test_name   the value of test name
      * @param string       $visit_label the value of visit label
      * @param string       $candidate   the value of candidate
+     * @param string       $site        the value of site
+     * @param string       $project     the value of project
```
Comment (driusan) [inline modules/data_team_helper/php/NDB_Form_data_team_helper.class.inc]: are you sure these are strings? It looks like they're being in the where column for SQL columns that are integers, not strings.

Can you also make the description more descriptive? What is "the value of site"? Is it the id? Is it the name? The abbreviation?

File: modules/data_team_helper/php/NDB_Form_data_team_helper.class.inc [hunk 10]
```diff
@@ -413,12 +487,16 @@ class NDB_Form_Data_Team_Helper extends NDB_Form
...
```
Comment (driusan) [inline modules/data_team_helper/php/NDB_Form_data_team_helper.class.inc]: same

File: modules/data_team_helper/php/NDB_Form_data_team_helper.class.inc [hunk 11]
```diff
@@ -466,11 +552,17 @@ class NDB_Form_Data_Team_Helper extends NDB_Form
...
```
Comment (driusan) [inline modules/data_team_helper/php/NDB_Form_data_team_helper.class.inc]: same

File: modules/data_team_helper/php/NDB_Form_data_team_helper.class.inc [hunk 12]
```diff
@@ -516,11 +616,17 @@ class NDB_Form_Data_Team_Helper extends NDB_Form
...
```
Comment (driusan) [inline modules/data_team_helper/php/NDB_Form_data_team_helper.class.inc]: same

File: modules/data_team_helper/php/NDB_Form_data_team_helper.class.inc [hunk 13]
```diff
@@ -567,8 +681,8 @@ class NDB_Form_Data_Team_Helper extends NDB_Form
      * Returns the percent completed for the given instrument
      *
      * @param unknown_type $instrument  the value of instrument
-     * @param null         $visit_label the value of visit lable
-     * @param null         $candidate   the value of candidate
+     * @param string       $visit_label the value of visit lable
+     * @param string       $candidate   the value of candidate
```
Comment (driusan) [inline modules/data_team_helper/php/NDB_Form_data_team_helper.class.inc]: same

(I'm not sure why these lines are changed in this PR since you're not using them or changing them, but if you're going to fix them they should be fixed right.)

**Comment by sruthymathew123** [comment]
@driusan Done.

**Comment by driusan** [comment]
I can't add a 🎉  icon on the message saying I finally merged this :(
